sph/include/sph/kernel_ve/momentum_energy_kern.hpp

```diff
@@ -192,16 +194,16 @@ momentumAndEnergyJLoop(int i, T sincIndex, T K, const cstone::Box<T>& box, const
             // T a = Wi * (2.0 * mj_pro_i + viscosity_ij * mi_roi);
             // T b = viscosity_ij * mj_roj_Wj;
 
-            energy += a_mom * T(2) * proi * (vx_ij * termA1_i + vy_ij * termA2_i + vz_ij * termA3_i);
+            energy += a_mom * proi * (vx_ij * termA1_i + vy_ij * termA2_i + vz_ij * termA3_i);
             // energy += vx_ij * (a * termA1_i + b * termA1_j) + vy_ij * (a * termA2_i + b * termA2_j) +
             //           vz_ij * (a * termA3_i + b * termA3_j);
         }
     }
 
     // with the choice of calculating coordinate (r) and velocity (v_ij) differences as i - j,
     // we add the negative sign only here at the end instead of to termA123_ij in each iteration
-    a_visc_energy = std::max(T(0), a_visc_energy);
-    du[i]         = T(0.5) * (energy + a_visc_energy);
+    a_visc_energy = (a_visc_energy > 0) ? a_visc_energy : T(0);
+    du[i]         = energy + T(0.5) * a_visc_energy;
```

Comment:
Add a comment stating that the 2 from 2P/rho has been simplified here, or we might add it by mistake in the future.

Reply:
will do and transfer to the new PR